odoo/fields.py

```diff
@@ -3452,7 +3452,7 @@ def write(self, records, value):
                 definition.pop('definition_changed', None)
 
             # update the properties definition on the container
-            container = records[self.definition_record]
+            container = records[self.definition_record].sudo()
```

Comment:
I'm ok for the first sudo, but not for this one, here you allow anyone to modify the definition

the definition edition should be disabled if we can't write on the parent (the traceback in the task is only about reading, so this line should not be the issue)

Reply:
agree to this.